akka-actor/src/main/scala/akka/io/TcpConnection.scala

```diff
@@ -150,11 +151,11 @@ private[io] abstract class TcpConnection(val tcp: TcpExt, val channel: SocketCha
     case write: WriteCommand ⇒
       if (writingSuspended) {
         if (TraceLogging) log.debug("Dropping write because writing is suspended")
-        sender() ! write.failureMessage
+        sender() ! write.failureMessage.setCause(DroppingWriteBecauseWritingIsSuspendedException)
```

Comment:
it looks scary to mutate an incoming message, can we add some sanity to it by having `withCause` that creates a copy and sets the cause, then the `setCause` can be `private`.